tardis/montecarlo/formal_integral.py

```diff
@@ -101,6 +103,29 @@ def make_source_function(self):
         plasma = self.plasma
         runner = self.runner
         atomic_data = self.plasma.atomic_data
+        macro_ref = atomic_data.macro_atom_references
+        macro_data = atomic_data.macro_atom_data
+
+        no_lvls = len(atomic_data.levels)
+        no_shells = len(model.w)
+
+        if runner.line_interaction_type == 'macroatom':
+            ma_int_data = macro_data.query('transition_type >=0')
+            internal_jump_mask = (macro_data.transition_type >= 0).values
+            internal = plasma.transition_probabilities[internal_jump_mask]
+
+            source_level_index = pd.MultiIndex.from_arrays([
+                    ma_int_data['atomic_number'].values,
+                    ma_int_data['ion_number'].values,
+                    ma_int_data['source_level_number'].values]
+            )
```

Comment:
that exists already in the atomic_data.macro_atom_data. it is called source_level_idx and destination_level_idx (I think the first one might not exits - but we should but it there).